keras_cv/src/models/object_detection/faster_rcnn/faster_rcnn.py

```diff
@@ -0,0 +1,477 @@
+import tree
+
+from keras_cv.src import layers as cv_layers
+from keras_cv.src.api_export import keras_cv_export
+from keras_cv.src.backend import keras
+from keras_cv.src.backend import ops
+from keras_cv.src.bounding_box.converters import _decode_deltas_to_boxes
+from keras_cv.src.bounding_box.utils import _clip_boxes
+from keras_cv.src.layers.object_detection.box_matcher import BoxMatcher
+from keras_cv.src.layers.object_detection.roi_align import _ROIAligner
+from keras_cv.src.layers.object_detection.roi_generator import ROIGenerator
+from keras_cv.src.layers.object_detection.roi_sampler import _ROISampler
+from keras_cv.src.layers.object_detection.rpn_label_encoder import (
+    _RpnLabelEncoder,
+)
+from keras_cv.src.models.object_detection.__internal__ import unpack_input
+from keras_cv.src.models.object_detection.faster_rcnn import FeaturePyramid
+from keras_cv.src.models.object_detection.faster_rcnn import RCNNHead
+from keras_cv.src.models.object_detection.faster_rcnn import RPNHead
+from keras_cv.src.models.task import Task
+from keras_cv.src.utils.train import get_feature_extractor
+from keras_cv.src import losses
+
+BOX_VARIANCE = [0.1, 0.1, 0.2, 0.2]
+
+
+@keras_cv_export(
+    [
+        "keras_cv.models.FasterRCNN",
+        "keras_cv.models.object_detection.FasterRCNN",
+    ]
+)
+class FasterRCNN(Task):
+    def __init__(
+        self,
+        backbone,
+        num_classes,
+        bounding_box_format,
+        anchor_generator=None,
+        feature_pyramid=None,
+        rcnn_head=None,
+        label_encoder=None,
+        *args,
+        **kwargs,
+    ):
+        # 1. Backbone
+        extractor_levels = ["P2", "P3", "P4", "P5"]
+        extractor_layer_names = [
+            backbone.pyramid_level_inputs[i] for i in extractor_levels
+        ]
+        feature_extractor = get_feature_extractor(
+            backbone, extractor_layer_names, extractor_levels
+        )
+        
+        # 2. Feature Pyramid
+        feature_pyramid = feature_pyramid or FeaturePyramid()
+        
+        # 3. Anchors
+        scales = [2**x for x in [0]]
```

Comment:
This just evaluates to: `scales  = [1]` always
Was that intentional?

Reply:
We can make dynamically using user specific inputs, generally retinanet which is a single stage has 3 scales.